src/sidebar/helpers/thread-sorters.js

```diff
@@ -89,11 +90,25 @@ export const sorters = {
     }
     const aLocation = location(a.annotation);
     const bLocation = location(b.annotation);
-    if (aLocation < bLocation) {
+
+    // If these annotations come from an EPUB and specify which chapter they
+    // came from via a CFI, compare the chapter order first.
+    if (aLocation.cfi && bLocation.cfi) {
+      const cfiResult = compareCFIs(aLocation.cfi, bLocation.cfi);
+      if (cfiResult !== 0) {
```

Comment:
Would you be so kind as to add a comment here that a result of `0` is a signal that the two annotations are in the same chapter? Probably seems obvious to you, and it is implied by a comment later on, but woulda saved me 5-10 minutes of grok time :)

Reply:
Makes sense. I've added a note.